scripts/Modules/Saritasa.Cake/Tasks/PackTask.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using Cake.Common.Diagnostics;
+using Cake.Common.IO;
+using Cake.Common.Tools.DotNet;
+using Cake.Common.Tools.DotNet.Build;
+using Cake.Frosting;
+
+namespace Saritasa.Cake;
+
+/// <summary>
+/// Pack task.
+/// </summary>
+[TaskName("Pack")]
+[TaskDescription("Build the library, test it and prepare nuget packages")]
+[IsDependentOn(typeof(CleanTask))]
+public sealed class PackTask : FrostingTask<PackContext>
+{
+    public const string AssemblyVersionAttribute = "AssemblyVersion";
+    public const string AssemblyFileVersionAttribute = "AssemblyFileVersion";
+    public const string AssemblyInformationalVersionAttribute = "AssemblyInformationalVersion";
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    public PackTask()
+    {
+        projectNamesForPack = new[]
+        {
+            "Saritasa.Tools.Common",
+            "Saritasa.Tools.Domain",
+            "Saritasa.Tools.EntityFrameworkCore",
+            "Saritasa.Tools.Emails",
+            "Saritasa.Tools.Misc",
+        };
+    }
+
+    private string[] projectNamesForPack;
+
+    /// <inheritdoc />
+    public override void Run(PackContext context)
+    {
+        UpdateAssemlyVersions(context);
+
+        PackProjects(context);
+
+        RevertAssemlyVersions(context);
+    }
+
+    private void UpdateAssemlyVersions(PackContext context)
+    {
+        var revcount = context.ExecuteGitCommand("rev-list --all --count").Replace(Environment.NewLine, string.Empty);
```

Comment:
Just curious, can we use https://github.com/cake-contrib/Cake_Git to work with git? That way we will not require having git installed and it is high level wrapper.

Reply:
I trying to use this extension for our purposes for a long time, but I did not find the necessary functionality there.

The main problem I have encountered is that it is not possible to call commands with custom options. I was able to partially replace the command ("git log --pretty=format:%h -n 1" to [GitLog](https://cakebuild.net/api/Cake.Git/GitAliases/469FCAAF)), but I do not think we should share the logic of working with git, like some commands executed via CakeGit, and others via CMD.

Reply:
`git log --pretty=format:%h -n 1`  - in fact this is just the first 7 digits of the first commit. It's ok to use cmd command if it is not easy to use extension.